src/examples/color-swatch/ColorSwatchCustomTrigger.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState } from 'react';
+import { ColorSwatchDialog } from '@zendeskgarden/react-colorpickers';
+import { Row, Col } from '@zendeskgarden/react-grid';
+import { Tooltip } from '@zendeskgarden/react-tooltips';
+import { IconButton } from '@zendeskgarden/react-buttons';
+import { ReactComponent as PaletteIcon } from '@zendeskgarden/svg-icons/src/16/palette-fill.svg';
+import { convertToMatrix } from '@zendeskgarden/container-utilities';
+import { PALETTE } from '@zendeskgarden/react-theming';
+
+interface IPaletteIconButton {
+  iconColor: string;
+}
+interface ILabledColor {
+  label: string;
+  value: string;
+}
+
+const colors = ['blue', 'red', 'yellow', 'green'];
+
+const shades = [200, 300, 400, 500, 600, 700, 800];
+
+const capitalize = (string: string) => string.charAt(0).toUpperCase() + string.slice(1);
+
+const labeledColors = colors.reduce(
+  (acc: ILabledColor[], hue: string) => [
+    ...acc,
+    ...shades.map(shade => ({
+      label: `${capitalize(hue)}-${shade}`,
+      value: (PALETTE as any)[hue][shade]
+    }))
+  ],
+  []
+);
```

Comment:
This is slick, but I honestly think the straight-up array constant you had before was a lot more readable (just needed to use `PALETTE` rather than hex string values). With this example code, we don't want users to get sucked into considering implementations for data structures unrelated to the use of the component. The idea is to drive them _past_ this kind of thing into the meat of the component usage.